extras/foreman_protector/dnf/foreman-protector.py

```diff
@@ -0,0 +1,77 @@
+import dnf
+import dnf.exceptions
+from dnfpluginscore import _, logger
+
+import configparser
+
+class ForemanProtector(dnf.Plugin):
+    name = 'foreman-protector'
+    config_name = 'foreman-protector'
+
+    def __init__(self,base,cli):
+        self.base = base
+        self.cli = cli
+
+    def config(self):
+        global fileurl
+
+        try:
+             parser = self.read_config(self.base.conf)
+        except Exception as e:
+            raise dnf.exceptions.Error(_("Parsing file failed: {}").format(str(e)))
+
+        if parser.has_section('main'):
+            fileurl = parser.get('main', 'whitelist')
+        else:
+            raise dnf.exceptions.Error(_('Incorrect plugin configuration!'))
+
+    def _load_whitelist(self):
+        package_whitelist = set()
+        try:
+            if fileurl:
+                llfile = open(fileurl, 'r')
+                for line in llfile.readlines():
+                    if line.startswith('#') or line.strip() == '':
+                        continue
+
+                    package_whitelist.add(line.rstrip().lower())
+                llfile.close()
+        except IOError as e:
+            raise dnf.exceptions.Error('Unable to read Foreman protector"s configuration: %s' % e)
+        return package_whitelist
+
+    def _add_obsoletes(self):
+        package_whitelist = self._load_whitelist()
+        final_query = self.base.sack.query()
+        if package_whitelist:
+        #  If anything obsoletes something that we have whitelisted ... then
+        # whitelist that too.
+            whitelist_query = self.base.sack.query().filterm(name=package_whitelist)
+            obsoletes_query = self.base.sack.query().filterm(obsoletes=list(whitelist_query))
+
+            final_query = whitelist_query.union(obsoletes_query)
+        return final_query
+
+    def sack(self):
+        whitelist_and_obsoletes = self._add_obsoletes()
+        all_available_updates = self.base.sack.query().available()
```

Comment:
Minor - the variable name `all_available_updates` is misleading. It contains all packages available in repositories, not only updates.

Reply:
corrected!